src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -2912,33 +2954,42 @@ public void resetLatestFlagAsFalse(List<ADTask> adTasks) {
         if (adTasks == null || adTasks.size() == 0) {
             return;
         }
-        BulkRequest bulkRequest = new BulkRequest();
+        List<BulkOperation> operations = new ArrayList<>();
         adTasks.forEach(task -> {
             try {
+
                 task.setLatest(false);
                 task.setLastUpdateTime(Instant.now());
-                IndexRequest indexRequest = new IndexRequest(DETECTION_STATE_INDEX)
-                    .id(task.getTaskId())
-                    .source(task.toXContent(XContentBuilder.builder(XContentType.JSON.xContent()), XCONTENT_WITH_TYPE));
-                bulkRequest.add(indexRequest);
+
+                BulkOperation op = new BulkOperation.Builder()
+                    .index(b -> b.index(DETECTION_STATE_INDEX).id(task.getTaskId()).document(task))
+                    .build();
+                operations.add(op);
             } catch (Exception e) {
                 logger.error("Fail to parse task AD task to XContent, task id " + task.getTaskId(), e);
             }
         });
 
-        bulkRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
-        client.execute(BulkAction.INSTANCE, bulkRequest, ActionListener.wrap(res -> {
-            BulkItemResponse[] bulkItemResponses = res.getItems();
-            if (bulkItemResponses != null && bulkItemResponses.length > 0) {
-                for (BulkItemResponse bulkItemResponse : bulkItemResponses) {
-                    if (!bulkItemResponse.isFailed()) {
-                        logger.warn("Reset AD tasks latest flag as false Successfully. Task id: {}", bulkItemResponse.getId());
+        BulkRequest bulkRequest = new BulkRequest.Builder().operations(operations).refresh(Refresh.True).build();
+
+        try {
+            CompletableFuture<BulkResponse> bulkResponse = sdkJavaAsyncClient.bulk(bulkRequest);
+            BulkResponse res = bulkResponse.orTimeout(10L, TimeUnit.SECONDS).get();
+
+            List<BulkResponseItem> bulkItemResponses = res.items();
+            if (bulkItemResponses != null && bulkItemResponses.size() > 0) {
+                for (BulkResponseItem bulkItemResponse : bulkItemResponses) {
+                    if (bulkItemResponse.error() == null) {
```

Comment:
Is `error` of Java Client equivalent to `isFailed` for RestClient?

Reply:
There is no direct equivalent for `isFailed` for the class `BulkIResponseItem`. However `.error()` returns the `ErrorCause` if any, or null. Therefore by checking if the `ErrorCause` is null, we can determine if there was an error.